source/getting-started/core-concepts.md

```diff
@@ -1,58 +1,120 @@
+  return `Mrs. ${firstName} ${secondName}`;
+});
+```
+
+```handlebars
+{{full-name firstName "Doe"}}
+```
+
+You can read more about helpers in [this API documentation](http://emberjs.com/api/classes/Ember.Helper.html).
 
 ## Components
 
 Components are the primary way user interfaces are organized in Ember. They
-consist of two parts: a template, and a source file written in JavaScript that
-defines the component's behavior. For example, a blog application might have a
-component for displaying a list of blog posts called `all-posts`, and another
-component for displaying an individual post called `view-post`. If users can
-upvote a post, the `view-post` component might define a behavior like _when the
-user clicks the upvote button, increase the `vote` property's value by 1_.
+consist of two parts:
 
-## Controllers
+* A template, for example `app/templates/components/welcome-message.hbs`
+* Optionally a source file written in JavaScript, for example
+  `app/components/welcome-message.js`.
+* **Components must contain a - in their name**. More about that later.
 
-Controllers are very much like components, so much so that in future versions of
-Ember, controllers will be replaced entirely with components. At the moment,
-components cannot be routed to (see below), but when this changes, it will be
-recommended to replace all controllers with components.
+Components differ from helpers in that they can output not only a value, but
+also manage DOM. An important property of components is that they are **isolated**.
+For example, a component's DOM is called "shadow" DOM.
 
-## Models
+In this example, a component's template adds a div around some other DOM:
 
-Models represent _persistent state_. For example, a blog application would want
-to save the content of a blog post when a user publishes it, and so the blog
-post would have a model defining it, perhaps called the `Post` model. A model
-typically persists information to a server, although models can be configured to
-save to anywhere else, such as the browser's Local Storage.
+```app/templates/components/my-component.hbs
+<div class="inner">
+  {{yield}}
+</div>
+```
 
-## Routes
+```handlebars
+<span>Hello</span>
+<div class="outer">
+  {{#my-component}}
+    Hello there!
+  {{/my-component}}
+</div>
+```
 
-Routes load a controller and a template. They can also load one or more models
-to provide data to the controller that can then be displayed by the template.
-For example, an `all-posts` route might load all the blog posts from the `Post`
-model, load the `all-posts` controller, and render the `all-posts` template.
-Similarly, a `view-post` route might load the model for the blog post to be
-shown, load the `view-post` controller, and render the `view-post` template.
+Because components deal with DOM, they are also responsible for maintaining
+the bulk of an application's state. Additionally they handle events like
+clicks, keyUp, and other more semtantic actions.
```

Comment:
Semantic to your application, instead of DOM events
On 17 Aug 2015 12:43, "greyhwndz" <notifications@github.com> wrote:

> In source/getting-started/core-concepts.md
> <https://github.com/emberjs/guides/pull/556#discussion_r37179594>:
>
> >
> > -Routes load a controller and a template. They can also load one or more models
> > -to provide data to the controller that can then be displayed by the template.
> > -For example, an `all-posts` route might load all the blog posts from the `Post`
> > -model, load the `all-posts` controller, and render the `all-posts` template.
> > -Similarly, a `view-post` route might load the model for the blog post to be
> > -shown, load the `view-post` controller, and render the `view-post` template.
> > +Because components deal with DOM, they are also responsible for maintaining
> > +the bulk of an application's state. Additionally they handle events like
> > +clicks, keyUp, and other more semtantic actions.
>
> "semantic" actions?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/emberjs/guides/pull/556/files#r37179594>.
>